tools/tpm2_createpolicy.c

```diff
@@ -186,48 +196,65 @@ TPM_RC build_policy(createpolicypcr_ctx *ctx,
         return rval;
     }
 
-    // Get policy hash.
     rval = Tss2_Sys_PolicyGetDigest(ctx->sapi_context,
             ctx->policy_session->sessionHandle, 0, &ctx->policy_digest, 0);
     if (rval != TPM_RC_SUCCESS) {
         LOG_ERR("Failed Policy Get Digest\n");
         return rval;
     }
 
-    LOG_INFO("Created Policy Digest:\n");
+    printf("Policy Digest: ");
+    int i=0;
+    for(i=0; i<ctx->policy_digest.t.size; i++) {
+        printf("%02X", ctx->policy_digest.t.buffer[i] );
+    }
+    printf("\n");
+    
 
-    //save the policy buffer in a file for use later or print hex to stdout.
-    if (ctx->policy_file_flag) {
-        bool result = files_save_bytes_to_file(ctx->policyfile, (UINT8 *) &ctx->policy_digest.t.buffer,
-                ctx->policy_digest.t.size);
-        if (!result) {
-            LOG_ERR("Failed to save policy digest into file \"%s\"",
-                    ctx->policyfile);
-            return -1;
+    if(!REAL_SESSION) {
+        printf("TRIAL SESSION:\n");
```

Comment:
This if else block is impossible to follow for me. I see tons of code duplication. It looks like it its a trial versus real session, you just call flush. Do something like...

```
code;
they;
have in common;
likely output code;

if (!REAL_SESSION) {
    different thing a;
} else {
    different thing b;
}
```